sdk/tables/azure-data-tables/azure/data/tables/_table_client.py

```diff
@@ -0,0 +1,460 @@
+from ._entity import TableEntity
+from ._generated import AzureTable
+from ._generated.models import AccessPolicy, SignedIdentifier, TableProperties, QueryOptions
+from ._serialize import _get_match_headers, _add_entity_properties
+from ._shared.base_client import parse_connection_str
+from ._shared._table_client_base import TableClientBase
+
+from ._shared.request_handlers import serialize_iso
+from ._shared.response_handlers import process_table_error
+
+from ._version import VERSION
+
+
+from ._models import TableEntityPropertiesPaged, UpdateMode
+
+from ._shared.response_handlers import return_headers_and_deserialized
+
+
+class TableClient(TableClientBase):
+    """ :ivar str account_name: Name of the storage account (Cosmos or Azure)"""
+
+    def __init__(
+            self, account_url,  # type: str
+            table_name,  # type: str
+            credential=None,  # type: Union[str,TokenCredential]
+            **kwargs  # type: Any
+    ):
+        # type: (...) -> None
+        """Create TableClient from a Credential.
+
+        :param account_url:
+            A url to an Azure Storage account.
+        :type account_url: str
+        :param table_name: The table name.
+        :type table_name: str
+        :param credential:
+            The credentials with which to authenticate. This is optional if the
+            account URL already has a SAS token, or the connection string already has shared
+            access key values. The value can be a SAS token string, an account shared access
+            key, or an instance of a TokenCredentials class from azure.identity.
+        :type credential: Union[str,TokenCredential]
+
+        :returns: None
+        """
+        super(TableClient, self).__init__(account_url, table_name, credential=credential, **kwargs)
+        self._client = AzureTable(self.url, pipeline=self._pipeline)
+        self._client._config.version = kwargs.get('api_version', VERSION)  # pylint: disable=protected-access
+
+    @classmethod
+    def from_connection_string(
+            cls, conn_str,  # type: str
+            table_name,  # type: str
+            **kwargs  # type: Any
+    ):
+        # type: (...) -> TableClient
+        """Create TableClient from a Connection String.
+
+        :param conn_str:
+            A connection string to an Azure Storage or Cosmos account.
+        :type conn_str: str
+        :param table_name: The table name.
+        :type table_name: str
+        :returns: A table client.
+        :rtype: ~azure.data.tables.TableClient
+        """
+        account_url, secondary, credential = parse_connection_str(
+            conn_str=conn_str, credential=None, service='table')
+        if 'secondary_hostname' not in kwargs:
+            kwargs['secondary_hostname'] = secondary
+        return cls(account_url, table_name=table_name, credential=credential, **kwargs)  # type: ignore
+
+    @classmethod
+    def from_table_url(cls, table_url, credential=None, **kwargs):
```

Comment:
Do we have a test for calling this constructor with a URL from the portal?

Reply:
in test_table_client yes!

Reply:
yes